packages/dashboard/src/components/three-fiber/camera-control.tsx

```diff
@@ -0,0 +1,63 @@
+import { useFrame, useThree } from '@react-three/fiber';
+import { OrbitControls } from 'three/examples/jsm/controls/OrbitControls';
+import React, { useEffect, useRef } from 'react';
+import { MOUSE, Vector3 } from 'three';
+
+interface CameraControlProps {
+  zoom: number;
+}
+
+export const updateZoom = (currentZoom: number, increment: number) =>
+  Math.max(0, Math.min(Infinity, currentZoom + increment));
+
+export const CameraControl: React.FC<CameraControlProps> = ({ zoom }) => {
+  const { camera, gl } = useThree();
+  const controlsRef = useRef<OrbitControls | null>(null);
+
+  useEffect(() => {
+    const controls = new OrbitControls(camera, gl.domElement);
+    controls.target = new Vector3(0, 0, -1000);
+    controls.enableRotate = false;
+    controls.enableDamping = false;
+    controls.enableZoom = true;
+    controls.mouseButtons = {
+      LEFT: MOUSE.PAN,
+      MIDDLE: undefined,
+      RIGHT: undefined,
+    };
+
+    controlsRef.current = controls;
+
+    // [CR]: Remove the icons for now. Code is left for future implementation.
+
+    // camera.zoom = zoom;
+    // camera.updateProjectionMatrix();
+
+    // const handleWheel = (event: WheelEvent) => {
+    //   const delta = event.deltaY;
+    //   const sensitivity = 0.1;
+
+    //   setInternalZoom((prevZoom) => {
+    //     const newZoom = prevZoom - delta * sensitivity;
+    //     return Math.max(controls.minZoom, Math.min(controls.maxZoom, newZoom));
+    //   });
+
+    //   updateExternalStateDebounced(internalZoom);
```

Comment:
I see that `debounce` is used here as well, I think when we implement it later on, we can just re-write it with a different callback instead of using a template. (if we still need it in the later implementation)

Reply:
Yes, definitely.
Do you suggest removing the commented code?

Reply:
oh sorry, I didn't mention what the action item was. I think it is ok as it is for now, just commented to make sure we understand what to do with debouncing later on